core/api/src/main/java/edu/uci/ics/texera/api/field/DateField.java

```diff
@@ -18,21 +18,26 @@
 
 
     public DateField(Date date) {
-        checkNotNull(date);
-        this.localDate = date.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+        if (date != null) {
+            this.localDate = date.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+        } else {
+            this.localDate = null;
+        }
     }
 
     public DateField(LocalDate localDate) {
-        checkNotNull(localDate);
         this.localDate = localDate;
     }
 
     @JsonCreator
     public DateField(
             @JsonProperty(value = JsonConstants.FIELD_VALUE, required = true) 

```

Comment:
@YinanDanielZhou remove `required = true` from all the JsonCreator constructors. Since they can be null so they are not required anymore.

Reply:
all removed